From e81ae2377196a2fcfe8847f8075c3a6fdc3b758a Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 31 Aug 2020 14:38:43 +0000
Subject: [PATCH] Conda Build fixed

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 141db72..10f332c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -142,7 +142,7 @@ find_package(Backtrace)
 if(Backtrace_FOUND)
   set(DMLC_EXECINFO_H_PRESENT 1)
   set(DMLC_EXECINFO_H ${Backtrace_HEADER})
-  target_include_directories(dmlc SYSTEM PRIVATE ${Backtrace_INCLUDE_DIRS})
+  target_include_directories(dmlc PRIVATE ${Backtrace_INCLUDE_DIRS})
   target_link_libraries(dmlc PRIVATE ${Backtrace_LIBRARIES})
 else()
   set(DMLC_EXECINFO_H_PRESENT 0)
-- 
1.8.3.1

