diff --git a/cmake/Utils.cmake b/cmake/Utils.cmake
index 963c494c..543a9d3d 100644
--- a/cmake/Utils.cmake
+++ b/cmake/Utils.cmake
@@ -91,7 +91,9 @@ function(format_gencode_flags flags out)
   # Set up architecture flags
   if(NOT flags)
     if (CUDA_VERSION VERSION_GREATER_EQUAL "11.1")
-      set(flags "52;60;61;70;75;80;86")
+      #set(flags "52;60;61;70;75;80;86")
+      string(REPLACE "," ";" tmpflags "$ENV{cuda_levels}")
+      string(REPLACE "." "" flags "${tmpflags};8.6")
     elseif (CUDA_VERSION VERSION_GREATER_EQUAL "11.0")
       set(flags "52;60;61;70;75;80")
     elseif(CUDA_VERSION VERSION_GREATER_EQUAL "10.0")
@@ -101,6 +103,7 @@ function(format_gencode_flags flags out)
     else()
       set(flags "35;50;52;60;61")
     endif()
+    message(STATUS "Using Open-CE Centralized CUDA Capabilities: ${flags}")
   endif()

   if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.18")
