diff --git a/python-package/xgboost/core.py b/python-package/xgboost/core.py
index 3752d93f..bb639aae 100644
--- a/python-package/xgboost/core.py
+++ b/python-package/xgboost/core.py
@@ -2196,7 +2196,7 @@ class Booster(object):
 
     def num_features(self) -> int:
         '''Number of features in booster.'''
-        features = ctypes.c_int()
+        features = c_bst_ulong()
         assert self.handle is not None
         _check_call(_LIB.XGBoosterGetNumFeature(self.handle, ctypes.byref(features)))
         return features.value
